### ISSUE TITLE
Provide OS Platform

### ISSUE THREAD

**Issue by thompson-tomo** [open-telemetry/opentelemetry-dotnet-contrib#1958]
### Component

OpenTelemetry.Resources.OperatingSystem

### Is your feature request related to a problem?

.

### What is the expected behavior?

Introduction of a new field -> os.platform which is a mapping of https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.isosplatform?view=net-8.0#system-runtime-interopservices-runtimeinformation-isosplatform(system-runtime-interopservices-osplatform)

### Which alternative solutions or features have you considered?

N/a

### Additional context

N/a

**Comment by Kielek** [comment]
`os.platform` is not defined in https://github.com/open-telemetry/semantic-conventions/blob/v1.26.0/docs/resource/os.md. It cannot be implemented without changes in sematnic convetions.
`os.type` is pretty close to this.

**Comment by thompson-tomo** [comment]
you are correct @Kielek  i missed submitting a Semantic convention issue, hence have now gone and created: https://github.com/open-telemetry/semantic-conventions/issues/1258 

Note for me the key difference for me is that, os platform are a group of Os types ie Linux.
